src/CommandAnalysis.cc

```diff
@@ -463,13 +467,63 @@ void CommandAnalysis::evaluate(const MemorySpecification& memSpec,
       if (mem_state != CommandAnalysis::MS_SREF) {
         cerr << "Incorrect use of Self-Refresh Power-Up!" << endl;
       }
-      if (max(zero, timestamp - sref_cycle) >= memSpec.memTimingSpec.RFC) {
-        sref_cycles         += max(zero, timestamp - sref_cycle
-                                   - memSpec.memTimingSpec.RFC);
+
+      // The total duration of self-refresh is given by the difference between
+      // the current clock cycle and the clock cycle of entering self-refresh.
+      int64_t sref_duration = timestamp - sref_cycle;
+
+      // Negative or zero duration should never happen.
+      assert(sref_duration > 0);
```

Comment:
Personally, I don't see the assertion as a punishment, but as a helper. As far I can see, people are only interested in valid traces. If this assumption holds, the assertion will ensure that the self-refresh duration is greater than zero (it may be possible that the user fixes the trace for his own benefit). The same principle applies to the second assertion inserted.

However, if you have a strong preference for warnings instead of assertions just let me know.